### PR TITLE
Standard Credentials File

### PR DESCRIPTION
**Pull request by tedivm**
This PR adds a standard credentials file to be used by third party applications.

### CODE REVIEW

File: SS3-Unified_Credentials_File.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+* `secure` should be set to `true` if SSL is available.
+* `token` should be set to the authorization token (currently only supported on the public server).
+* `username` is the screeps username, and is only supported on private servers.
+* `password` is the screeps password, and is only supported on private servers.
+
+
+## Full Example
+
+```yaml
+connections:
```
Comment (AlinaNova21) [inline SS3-Unified_Credentials_File.md:49]: According to line 28 above, this should be `servers`

**Comment by AlinaNova21** [comment]
As a side thought, most screeps utilities have other options along side the connection, for example, screeps console has `max_history`, `max_scroll`, `smooth_scroll`, etc. Would it be worth adding a `configs` section that allows each apps config to be specified? These could easily be overridden by a local config for cases such as multiple stats agents.
ex: 
```yaml
servers:
  ...
configs:
  screepsconsole:
    max_history: 20000
    max_scroll: 20000
  screepsplus-agent:
    token: sptoken
    checkForUpdates: false
```

**Comment by Jomik** [comment]
As I mentioned to @ags131 in slack, I propose something more along the lines of

* Project Root (Optional) - (`project/.screeps.yaml`)
* Current Working Directory - (`./.screeps.yaml`)
* Screeps config env - (`$SCREEPS_CONFIG`)
* XDG Config Home - (`${XDG_CONFIG_HOME:-$HOME/.config}/screeps/config.yaml`)
* Home Directory - (`~/.screeps.yaml`)

**Comment by AlinaNova21** [comment]
* Added the `SCREEPS_CONFIG` env variable option, I put it first in the list since it should override any other paths. 
* Adjusted `$XDG_CONFIG_HOME` to use variable rather than `~/.config` ,
* Added `%APPDATA%` for windows users

IMO, any where the env variable isn't set should be skipped in implementations. Thus the default (`${XDG_CONFIG_HOME:-$HOME/.config}`) isn't needed

**Comment by tedivm** [comment]
@ags131 if you think this is ready to merge lets go ahead and do it.

**Comment by Jomik** [comment]
Well, my only thing is that the spec for xdg says to use `${XDG_CONFIG_HOME:-$HOME/.config}`
That is, if it is not set, then fallback to `~/.config`, my system does not have the env variable set, and I do use xdg config home..
But I mean, it is rather simple to set, just inconsistent with the spec.

**Comment by tedivm** [comment]
Yeah, the xdg spec does have a default for if the environmental variable is set, so it does make sense to fall back on that. I think the clearest way to do that would be to explicitly add the default xdg location to the list (after the one where the environmental variable is set)-

```
* XDG Config Default Directory - (`$HOME/.config/screeps/config.yaml`)
```

**Comment by AlinaNova21** [comment]
Thats easy enough to add, For windows, should these paths still be checked or just skipped? (`$HOME` doesn't exist, instead its `%USERPROFILE%`)

**Comment by Jomik** [comment]
Well, I guess xdg doesn't make sense on Windows though.

**Comment by AlinaNova21** [comment]
I'm fine implementing that as just if !Windows, add XDG paths

**Comment by AlinaNova21** [comment]
LGTM to merge. @Jomik, any other concerns before merging?  (We can always open other PRs later to discuss changes too :) )

**Comment by Jomik** [comment]
I'm happy with it for my uses :grin:

**Pull request merged**